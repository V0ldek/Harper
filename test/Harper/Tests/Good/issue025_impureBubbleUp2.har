ref RefBox a = {
    data = {
        a :: a;
    }

    ctor :: a -> RefBox a;
    ctor a = {
        this.data.a := a;
        return this;
    };

    get :: impure -> a;
    get = this.data.a;

    set :: a -> impure -> ();
    set a = {
        this.data.a := a;
    };
};

f :: RefBox Integer -> sideeffect -> Integer;
f b = {
    eval printLn "42" () ;
    return b.get ();
};

main :: sideeffect -> Integer;
main = {
    (b :: RefBox Integer) = RefBox.ctor 42;
    return f b ();
};