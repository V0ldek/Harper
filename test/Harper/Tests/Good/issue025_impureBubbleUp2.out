42

Execution ended with value: 42